File: uithelp/backend/src/modules/comments/comment.model.ts
import mongoose from "mongoose";

const CommentSchema = new mongoose.Schema(
    {
        postId: { type: String, required: true },
        userId: { type: String, required: true },
        userName: { type: String, default: "" },
        content: { type: String, required: true },
        isDeleted: { type: Boolean, default: false },
        deletedAt: { type: Date, default: null },
    },
    {
        timestamps: true
    }
);

CommentSchema.set("toJSON", {
    transform: (doc, ret) => {
        const r = ret as any; // 👈 tránh lỗi TS

        const formatVN = (date: Date) =>
            new Date(date).toLocaleString("vi-VN", {
                timeZone: "Asia/Ho_Chi_Minh",
            });

        if (r.createdAt) r.createdAtVN = formatVN(r.createdAt);
        if (r.updatedAt) r.updatedAtVN = formatVN(r.updatedAt);

        return r;
    },
});

export default mongoose.model("Comment", CommentSchema);

File: uithelp/frontend/lib/data/models/comment_model.dart
class CommentModel {
  final String id;
  final String postId;
  final String userId;
  final String userName;
  final String content;
  final DateTime createdAt;

  const CommentModel({
    required this.id,
    required this.postId,
    required this.userId,
    required this.userName,
    required this.content,
    required this.createdAt,
  });

  factory CommentModel.fromJson(Map<String, dynamic> json) => CommentModel(
    id: json['_id'] as String? ?? '',
    postId: json['postId'] as String? ?? '',
    userId: json['userId'] as String? ?? '',
    userName: json['userName'] as String? ?? '',
    content: json['content'] as String? ?? '',
    createdAt: json['createdAt'] != null
        ? DateTime.tryParse(json['createdAt']) ?? DateTime.now()
        : DateTime.now(),
  );
}
